### PR TITLE
Register application actions in the Audit Log #9316

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import com.enonic.xp.data.PropertyTree;
+import com.enonic.xp.security.PrincipalKey;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+
+@Component(configurationPid = "com.enonic.xp.app")
+public class ApplicationAuditLogSupportImpl
+    implements ApplicationAuditLogSupport
+{
+    private static final String SOURCE = "com.enonic.xp.core-app";
```
Comment (rymsha) [inline modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java:27]:
```suggestion
    private static final String SOURCE = "com.enonic.xp.app";
```

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        if ( appConfig.auditlog_enabled() )
+        {
+            final PropertyTree data = new PropertyTree();
+
+            log( "system.application.install", data, AuditLogUris.from( applicationKey.toString() ) );
+        }
+    }
+
+    @Override
+    public void installStoredApplication( final ApplicationKey applicationKey, final ApplicationInstallationParams params )
```
Comment (rymsha) [inline modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java:87]: this should not be logged ever.

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java (hunk 2 above)
Comment (anatol-sialitski) [reply modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java]: done

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java (hunk 1 above)
Comment (anatol-sialitski) [reply modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java]: done

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupport.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+public interface ApplicationAuditLogSupport
+{
+    void startApplication( ApplicationKey applicationKey );
+
+    void stopApplication( ApplicationKey applicationKey );
+
+    void installApplication( ApplicationKey applicationKey, URL url );
+
+    void installApplication( ApplicationKey applicationKey );
```
Comment (rymsha) [inline modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupport.java:15]: sure you need a separate method? I don't insist, just wonder.

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    private static final String SOURCE = "com.enonic.xp.app";
+
+    private final AuditLogService auditLogService;
+
+    private final AppConfig appConfig;
+
+    @Activate
+    public ApplicationAuditLogSupportImpl( final AppConfig appConfig, @Reference final AuditLogService auditLogService )
+    {
+        this.appConfig = appConfig;
```
Comment (rymsha) [inline modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java:35]: you can write use method with `@Modify` annotation. So service is not recreated when config is changed.

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        this.auditLogService = auditLogService;
+    }
+
+    @Override
+    public void startApplication( final ApplicationKey applicationKey )
+    {
+        if ( appConfig.auditlog_enabled() )
```
Comment (rymsha) [inline modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java:42]: at could be stored in `volatile boolean` for readability.

**Review by rymsha — COMMENTED** [review]
InstallApplicationParams triggerEvent (that is always false), should always be false. And builder method must be deprecated.

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupport.java (hunk 3 above)
Comment (anatol-sialitski) [reply modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupport.java:15]: We can do it using one method. But I don't like to pass the `null` argument to the method. As for me, in this case the explicit argument is more suitable and a developer will not waste a time to investigate what to do this method.

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java (hunk 4 above)
Comment (anatol-sialitski) [reply modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java]: done

File: modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java (hunk 5 above)
Comment (anatol-sialitski) [reply modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupportImpl.java]: done

**Review by rymsha — APPROVED** [review]

File: modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationInstallationParams.java [hunk 6]
```diff
@@ -11,7 +11,7 @@
     private ApplicationInstallationParams( final Builder builder )
     {
         start = builder.start;
-        triggerEvent = builder.triggerEvent;
+        triggerEvent = false;
```
Comment (vbradnitski) [inline modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationInstallationParams.java:14]: Do we need a field anymore? Wouldn't be better to return 'false' directly in getter?

**Review by vbradnitski — CHANGES_REQUESTED** [review]

File: modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationInstallationParams.java (hunk 6 above)
Comment (anatol-sialitski) [reply modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationInstallationParams.java:14]: It still needs for `hashCode` and `equals` methods.

**Review by vbradnitski — APPROVED** [review]

**Pull request merged**